src/main/java/io/cryostat/net/web/http/api/v2/CertificatePostHandler.java

```diff
@@ -166,9 +166,18 @@ public boolean isOrdered() {
         } catch (CertificateEncodingException cee) {
             throw new ApiException(500, cee.getMessage(), cee);
         } catch (Exception e) {
+            deleteMalformedCert(filePath, e);
```

Comment:
I think we can avoid this by restructuring the main body of the `try` above. Currently it looks like the output file is created right away in the `try-with-resources` when we do `new FileOutputStream`, so it looks like this:

```
var fileIn = readFile(uploadedCert);
var fileOut = writeFile(newCertPath);

var certs = certParser.parse(fileIn);
for (cert in certs) {
  fileOut.write(cert.encoded());
}
```

So, if parsing fails, we have opened and closed the `fileOut` without ever writing any content. We can reorganize it along the lines of:

```
var fileIn = readFile(uploadedCert);
var certs = certParser.parse(fileIn);

var fileOut = writeFile(newCertPath);
for (cert in certs) {
  fileOut.write(cert.encoded());
}
```

so that if we fail to parse the uploaded file then we never create the output file to begin with.